src/features/foldingProvider.ts

```diff
@@ -99,16 +99,43 @@ export default class AsciidocFoldingRangeProvider implements vscode.FoldingRange
     }
   }
 
+  private static handleSingleLineCommentFoldingRanges (singleLineCommentStartIndexes: any[], foldingRanges: any[], lineIndex: number, lineText: string,
+    isPreviousLineSingleCommented: boolean) {
```

Comment:
I don't think we need the `isPreviousLineSingleCommented` variable.
If the line starts with `//`, there are 2 cases:

- it's the beginning of one or more single line comments
- it's the continuation of single line comments

If the line does not start with `//`, there are 3 cases:

- it's the end of only one single line comment
- it's the end of two or more single line comments
- it's neither

**Logic**
- When the line starts with `//` and `singleLineCommentStartIndexes` is empty, we should add the current line index to it: `singleLineCommentStartIndexes.push(lineIndex)`.
- When the line starts with `//` and `singleLineCommentStartIndexes` is not empty, it's a continuation of single line comments (and we don't need to do anything).
- When the line does not start with `//` and `singleLineCommentStartIndexes` is empty, we don't need to anything.
- When the line does not start with `//` and `singleLineCommentStartIndexes` is not empty, we should check if the current line number is greater than `singleLineCommentStartIndexes.pop() + 1`. If that's the case, then we have two or more single line comments. If not, then it's a single line comment.

**Pseudo-code**
```js
if (lineText.startsWith('//')) {
  if (singleLineCommentStartIndexes.length === 0) {
    singleLineCommentStartIndexes.push(lineIndex)
  }
} else {
  if (singleLineCommentStartIndexes.length !== 0) {
    const startIndex = singleLineCommentStartIndexes.pop()
    const endIndex = lineIndex - 1
    if (endIndex > startIndex) {
      foldingRanges.push(new vscode.FoldingRange(
          startIndex,
          endIndex,
          FoldingRangeKind.Region))
      }
    }
  }
}
```